uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1092,18 +1092,23 @@ sub dicom_to_minc {
     my $this = shift;
     my ($study_dir, $converter,$get_dicom_info,
 		$exclude,$mail_user, $upload_id) = @_;
-    my ($d2m_cmd,$d2m_log,$exit_code);
+    my ($d2m_cmd, $d2m_log, $exit_code, $excluded_regex);
     my $message = '';
 
     # create the excluded series description regex necessary to exclude the
     # series description specified in the Config Setting
     # excluded_series_description
-    my $excluded_regex = join('|', map { quotemeta($_) } @$exclude);
+    if ($exclude && ref($exclude) eq 'ARRAY') {
+        $excluded_regex = join('\s+$|', map { quotemeta($_) }@$exclude) . '\s+$';
```

Comment:
I think what you really want is

```perl
$excluded_regex = join('|', map { "\t\Q$_\E$" } @$exclude)
```

basically I am taking all the elements of `@$exclude` and doing this:

1. Apply `quotemeta` on the element (note that `"\Q$var\E" = quotemeta($var)`)
2. Add a tab at the beginning
3. Add a `$` at the end

Once I've done this to all the elements in `@$exclude`, I build the regex by joining the elements with `|`.

This should check if there's a match in the *last* field in the (tab-separated) list reported by `get_dicom_info.pl`.

Makes sense?

Reply:
Thanks for the suggestion! I applied it in my last commit. The only thing is that I could not include the $ `{ "\t\Q$_\E$" }` as it got me an error and if I escape it, it is not what I want either. Not the prettiest but at least it works!